activiti-core/activiti-engine/src/main/java/org/activiti/engine/delegate/event/impl/ActivitiEventBuilder.java

```diff
@@ -526,19 +526,21 @@ public static ActivitiMembershipEvent createMembershipEvent(ActivitiEventType ty
         return newEvent;
     }
 
-    protected static void populateEventWithCurrentContext(ActivitiEventImpl event) {
+    protected static void populateEventWithCurrentContextAndActor(ActivitiEventImpl event, String actor) {
         if (event instanceof ActivitiEntityEvent) {
             Object persistedObject = ((ActivitiEntityEvent) event).getEntity();
-            if (persistedObject instanceof Job) {
-                event.setExecutionId(((Job) persistedObject).getExecutionId());
-                event.setProcessInstanceId(((Job) persistedObject).getProcessInstanceId());
-                event.setProcessDefinitionId(((Job) persistedObject).getProcessDefinitionId());
-            } else if (persistedObject instanceof DelegateExecution) {
-                event.setExecutionId(((DelegateExecution) persistedObject).getId());
-                event.setProcessInstanceId(((DelegateExecution) persistedObject).getProcessInstanceId());
-                event.setProcessDefinitionId(((DelegateExecution) persistedObject).getProcessDefinitionId());
-            } else if (persistedObject instanceof IdentityLinkEntity) {
-                IdentityLinkEntity idLink = (IdentityLinkEntity) persistedObject;
+            if (persistedObject instanceof Job job) {
+                event.setExecutionId(job.getExecutionId());
+                event.setProcessInstanceId(job.getProcessInstanceId());
+                event.setProcessDefinitionId(job.getProcessDefinitionId());
+            } else if (persistedObject instanceof DelegateExecution delegateExecution) {
+                event.setExecutionId(delegateExecution.getId());
+                event.setProcessInstanceId(delegateExecution.getProcessInstanceId());
+                event.setProcessDefinitionId(delegateExecution.getProcessDefinitionId());
+                if (event.getType().equals(ActivitiEventType.PROCESS_COMPLETED) && actor != null) {
```

Comment:
I don't think you need to check for the specific event type here. Checking if the actor is set it should be enough:
```suggestion
                if (actor != null) {
```